metaflow/metadata/service.py

```diff
@@ -102,19 +120,21 @@ def _get_object_internal(cls, obj_type, obj_order, sub_type, sub_order, filters=
                 return None
             raise
 
-    def _new_run(self, tags=[], sys_tags=[]):
+    def _new_run(self, run_id=None, tags=[], sys_tags=[]):
         # first ensure that the flow exists
         self._get_or_create('flow')
-        run = self._get_or_create('run', tags=tags, sys_tags=sys_tags)
+        run = self._get_or_create('run', run_id, tags=tags, sys_tags=sys_tags)
```

Comment:
We should improve the _get_or_create logic. Right now it does a query, waits for a 404 and creates it (which is already somewhat ugly).

Reply:
What do you propose instead?

Reply:
I would pass a always_create parameter as an additional optional param to _get_or_create that defaults to None. In _get_or_create, if that parameter is set, use that, otherwise use the logic inside to create. Also don't create on 404 to allow basically issues to be caught (basically: if we shoudln't be creating, it should be an error that the obj doesn't exist).

Reply:
We can make that change outside of this PR.